src/main/scala/com/databricks/spark/sql/perf/mllib/data/dataGeneration.scala

```diff
@@ -189,3 +213,47 @@ class GaussianMixtureDataGenerator(
   override def copy(): GaussianMixtureDataGenerator =
     new GaussianMixtureDataGenerator(numCenters, numFeatures, seed)
 }
+
+class RandStringGenerator(
+    val minLen: Int,
+    val maxLen: Int,
+    val charSet: String) extends RandomDataGenerator[String] {
+
+  private val rng = new java.util.Random()
+
+  override def nextValue(): String = {
+    val sb = new StringBuilder
+    var i = 0
+    val strLen = if (minLen == maxLen) minLen else {
+      minLen + rng.nextInt(maxLen - minLen + 1)
+    }
+    while (i < strLen) {
+      val nextChar = charSet.charAt(rng.nextInt(charSet.length))
+      sb.append(nextChar)
+      i += 1
+    }
+    sb.toString()
+  }
+
+  override def setSeed(seed: Long) {
+    rng.setSeed(seed)
+  }
+
+  override def copy(): RandStringGenerator = new RandStringGenerator(minLen, maxLen, charSet)
```

Comment:
@smurching I check again, here no need to reset seed, because in `RandomRDDs.randomRDD` it will reset the seed for each partition. (by call method `setSeed`)